src/vm/eventpipeconfiguration.cpp

```diff
@@ -35,12 +35,26 @@ EventPipeConfiguration::~EventPipeConfiguration()
     }
     CONTRACTL_END;
 
+    if(m_pConfigProvider != NULL)
+    {
+        delete(m_pConfigProvider);
+        m_pConfigProvider = NULL;
+    }
+
     if(m_pEnabledProviderList != NULL)
     {
         delete(m_pEnabledProviderList);
         m_pEnabledProviderList = NULL;
     }
 
+    SListElem<EventPipeProvider*> *pElem = m_pProviderList->GetHead();
+    while(pElem != NULL)
+    {
+        SListElem<EventPipeProvider*> *pCurElem = pElem;
+        pElem = m_pProviderList->GetNext(pElem);
```

Comment:
I think you'll also want to delete the provider itself as well as the list element.